src/strongbox/github_api.clj

```diff
@@ -55,127 +219,75 @@
-                  no-known-game-tracks? (do (debug (format "no game track detected for release '%s' and asset '%s', assuming 'retail'"
-                                                           (:name release) (:name asset)))
-                                            {:game-track :retail, :version version :-mo :sa--ngt})
-
-                  ;; no game track present in asset name or release name and just a single entry in `:game-track-list`. use that.
-                  single-game-track? {:game-track (first known-game-tracks), :version version,  :-mo :sa--1gt}
-
-                  ;; no game track present in asset name or release name with multiple entries in `:game-track-list`.
-                  ;; assume all entries in `:game-track-list` supported.
-                  many-game-tracks? (vec (for [game-track known-game-tracks]
-                                           {:game-track game-track, :version version, :-mo :sa--Ngt}))
-
-                  :else (error (format "unhandled state attempting to determine game track(s) for asset '%s' in release of '%s'"
-                                       asset addon)))
-
-                update-list (if (sequential? update-list) update-list [update-list])]
-
-            (mapv (fn [update]
-                    (merge asset update)) update-list)))]
-    (->> asset-list (map updater) flatten (group-by :game-track))))
-
-(defn-spec parse-github-release-data vector?
-  "given a `release-list` (a response from Github), parse the assets in each release."
-  [release-list vector?, addon :addon/expandable, game-track ::sp/game-track]
-  (->> release-list
-       (map (partial group-assets addon))
-       (filter #(contains? % game-track))
-       (map game-track)
-       vec))
-
-(defn-spec expand-summary (s/or :ok :addon/release-list, :error nil?)
-  "fetches a list of releases from the addon host for the given `addon-summary`"
-  [addon :addon/expandable, game-track ::sp/game-track]
-  (let [wrangle-release (fn [release]
-                          (when-let [asset (first release)]
-                            {:download-url (:browser_download_url asset)
-                             :version (:version asset)
-                             :game-track game-track}))
-        wrangle-release-list #(mapv wrangle-release %)]
-    (some-> addon
-            :source-id
-            download-releases
-            (parse-github-release-data addon game-track)
-            wrangle-release-list
-            utils/nilable)))
-
-;;
+(defn-spec find-gametracks-release-list (s/or :ok ::sp/game-track-list, :no-game-tracks nil?)
+  "analyses all releases to date and returns a list of game tracks found.
+  returns `nil` if no game tracks are detected.
+  does not guess, fast and cheap to do."
+  [release-list ::sp/list-of-maps]
+  (let [known-game-tracks []]
+    (->> release-list
+         (map-indexed (parse-assets* known-game-tracks))
```

Comment:
disentangle this. I don't want `known-game-tracks` to be an implicit 'don't fetch release.json trigger'.

Reply:
test